lite/examples/pose_estimation/android/app/src/main/java/org/tensorflow/lite/examples/poseestimation/ml/MoveNet.kt

```diff
@@ -0,0 +1,341 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================
+*/
+
+package org.tensorflow.lite.examples.poseestimation.ml
+
+import android.content.Context
+import android.graphics.*
+import android.os.SystemClock
+import org.tensorflow.lite.DataType
+import org.tensorflow.lite.Interpreter
+import org.tensorflow.lite.examples.poseestimation.data.*
+import org.tensorflow.lite.gpu.GpuDelegate
+import org.tensorflow.lite.support.common.FileUtil
+import org.tensorflow.lite.support.image.ImageProcessor
+import org.tensorflow.lite.support.image.TensorImage
+import org.tensorflow.lite.support.image.ops.ResizeOp
+import org.tensorflow.lite.support.image.ops.ResizeWithCropOrPadOp
+import org.tensorflow.lite.support.tensorbuffer.TensorBuffer
+import kotlin.math.abs
+import kotlin.math.max
+import kotlin.math.min
+
+enum class ModelType {
+    Lightning,
+    Thunder
+}
+
+class MoveNet(private val interpreter: Interpreter) : PoseDetector {
+
+    companion object {
+        private const val MIN_CROP_KEYPOINT_SCORE = .2f
+        private const val CPU_NUM_THREADS = 4
+
+        // allow specifying model type.
+        fun create(context: Context, device: Device, modelType: ModelType): MoveNet {
+            val options = Interpreter.Options()
+            options.setNumThreads(CPU_NUM_THREADS)
+            when (device) {
+                Device.CPU -> {
+                }
+                Device.GPU -> {
+                    // Movenet TFLite models don't work with GPUDelegate yet
+//                    options.addDelegate(GpuDelegate())
```

Comment:
This should have a TODO and a bug ID. Also, have we confirmed that the models are compatible with NNAPI for any of our Pixel devices? That is, that they are accelerated on hardware? Or are these float models (not yet quantized)?

Reply:
Filed bugs for tracking.